### PR TITLE
Improve reference resolution

### LINKED ISSUE #491: Fix renaming in lambda
**Describe the bug**

```candy
main := { _ ->
  foo = 123 # Rename doesn't work
  ✨.print foo # Rename works
}
```

### LINKED ISSUE #494: Fix references / go to definition inside text interpolation
**Describe the bug**

```candy
foo = 1
foo
"{foo}"
```

### PR DESCRIPTION
**Pull request by skyz1**
Closes: #491, Closes #494

An Id of a higher representation can get mapped to multiple Ids of a lower representation. Before this PR only one Id of the lower representation got returned in an inconsistent manner when the mapping was resolved. This PR changes the lookup to return all corresponding Ids. This allows the caller of the lookup to decide how to handle the Ids. 

These changes make definition lookups and renaming work consistently. Additionally reference resolution now also works in text templates.

### Checklist

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: compiler/cli/src/utils.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ pub fn packages_path() -> PackagesPath {
         .find(|path| path.ends_with("target"))
         .unwrap();
     let candy_repo = target_dir.parent().unwrap();
-    PackagesPath::try_from(candy_repo.join("packages").as_path()).unwrap()
+    PackagesPath::try_from(candy_repo.join("packages").join("Core").as_path()).unwrap()
```
Comment (JonasWanke) [inline compiler/cli/src/utils.rs]: The `PackagesPath` is the directory containing all packages that can be imported and not the path to a specific package like Core

```suggestion
    PackagesPath::try_from(candy_repo.join("packages").as_path()).unwrap()
```

File: compiler/language_server/src/features_candy/find_definition.rs [hunk 2]
```diff
@@ -19,8 +19,11 @@ pub fn find_definition(db: &Database, module: Module, offset: Offset) -> Option<
         _ => return None,
     }
 
-    let origin_hir_id = db.cst_to_hir_id(module.clone(), origin_cst.data.id)?;
-    let origin_expression = db.find_expression(origin_hir_id)?;
+    let origin_hir_ids = db.cst_to_hir_id(module.clone(), origin_cst.data.id);
+    let [origin_hir_id] = origin_hir_ids.as_slice() else {
+        panic!("The CST-Id of an Identifier should map to exactly one HIR-Id")
+    };
+    let origin_expression = db.find_expression(origin_hir_id.clone())?;
```
Comment (JonasWanke) [inline compiler/language_server/src/features_candy/find_definition.rs]: Destructuring definitely looks nicer, but we can avoid the `clone()`:

```suggestion
    assert_eq!(
        origin_hir_ids.len(),
        1,
        "The CST ID of an identifier should map to exactly one HIR ID.",
    );
    let origin_hir_id = origin_hir_ids.into_iter().next().unwrap();
    let origin_expression = db.find_expression(origin_hir_id)?;
```

File: compiler/language_server/src/features_candy/references.rs [hunk 3]
```diff
@@ -39,29 +39,35 @@ where
...
-                    match hir_expr {
-                        Expression::Reference(target_id) => target_id,
-                        Expression::Symbol(_) => {
-                            // TODO: Handle struct access
-                            return None;
+            let hir_id = db.cst_to_hir_id(module, origin_cst.data.id);
+            let [hir_id] = hir_id.as_slice() else {
+                panic!("The CST-Id of an Identifier should map to exactly one HIR-Id")
+            };
+            let hir_id = hir_id.to_owned();
```
Comment (JonasWanke) [inline compiler/language_server/src/features_candy/references.rs]:
```suggestion
            let hir_ids = db.cst_to_hir_id(module, origin_cst.data.id);
            assert_eq!(
                hir_ids.len(),
                1,
                "The CST ID of an identifier should map to exactly one HIR ID.",
            );
            let hir_id = hir_ids.into_iter().next().unwrap();
```

**Review by JonasWanke — CHANGES_REQUESTED** [review]

File: compiler/cli/src/utils.rs (hunk 1 above)
Comment (skyz1) [reply compiler/cli/src/utils.rs]: I thought this was a mistake. Without this change displaying the intermediate representation of core modules using the extension was broken for me. I checked again and the issue was actually a wrong packages path in my VS Code configuration.

**Review by JonasWanke — APPROVED** [review]

**Pull request merged**